x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/logic/diff/calculation/algorithms/kql_query_diff_algorithm.test.ts

```diff
@@ -0,0 +1,544 @@
+          },
+          target_version: {
+            type: KqlQueryType.inline_query,
+            query: 'query string = true',
+            language: KqlQueryLanguageEnum.kuery,
+            filters: [],
+          },
+        };
+
+        const result = kqlQueryDiffAlgorithm(mockVersions);
+
+        expect(result).toEqual(
+          expect.objectContaining({
+            has_base_version: false,
+            base_version: undefined,
+            merged_version: mockVersions.current_version,
+            diff_outcome: ThreeWayDiffOutcome.MissingBaseNoUpdate,
+            merge_outcome: ThreeWayMergeOutcome.Current,
+            conflict: ThreeWayDiffConflict.NONE,
+          })
+        );
+      });
+
+      it('returns current_version as merged output if if all versions are saved query types', () => {
+        const mockVersions: ThreeVersionsOf<RuleKqlQuery> = {
+          base_version: MissingVersion,
+          current_version: {
+            type: KqlQueryType.saved_query,
+            saved_query_id: 'saved-query-id',
+          },
+          target_version: {
+            type: KqlQueryType.saved_query,
+            saved_query_id: 'saved-query-id',
+          },
+        };
+
+        const result = kqlQueryDiffAlgorithm(mockVersions);
+
+        expect(result).toEqual(
+          expect.objectContaining({
+            has_base_version: false,
+            base_version: undefined,
+            merged_version: mockVersions.current_version,
+            diff_outcome: ThreeWayDiffOutcome.MissingBaseNoUpdate,
+            merge_outcome: ThreeWayMergeOutcome.Current,
+            conflict: ThreeWayDiffConflict.NONE,
+          })
+        );
+      });
+    });
+
+    describe('if current_version and target_version are different - scenario -AB', () => {
+      it('returns target_version as merged output if all versions are the same', () => {
+        const mockVersions: ThreeVersionsOf<RuleKqlQuery> = {
+          base_version: MissingVersion,
+          current_version: {
+            type: KqlQueryType.inline_query,
+            query: 'query string = true',
+            language: KqlQueryLanguageEnum.kuery,
+            filters: [],
+          },
+          target_version: {
+            type: KqlQueryType.inline_query,
+            query: 'query string = false',
+            language: KqlQueryLanguageEnum.kuery,
+            filters: [],
+          },
+        };
+
+        const result = kqlQueryDiffAlgorithm(mockVersions);
+
+        expect(result).toEqual(
+          expect.objectContaining({
+            has_base_version: false,
+            base_version: undefined,
+            merged_version: mockVersions.target_version,
+            diff_outcome: ThreeWayDiffOutcome.MissingBaseCanUpdate,
+            merge_outcome: ThreeWayMergeOutcome.Target,
+            conflict: ThreeWayDiffConflict.SOLVABLE,
+          })
+        );
+      });
+
+      it('returns target_version as merged output if all versions are different', () => {
```

Comment:
Not sure I understand these two scenarios as described here:

1. `returns target_version as merged output if all versions are the same`: Do you mean here `if current and target versions have the same type`?
2. `returns target_version as merged output if all versions are different`: Did you mean here `if current and target versions have different type`? In that case it doesn't match the input data, which are both of `saved_query` type.

Reply:
Yeah I meant if both versions had the same/different `type` fields, I'll switch it over 👍